packages/lit-dev-content/site/tutorials/content/first-component/04.md

```diff
@@ -0,0 +1,104 @@
+By the end of this step, the shown word will automatically change every second.
+
+This can be achieved by using
+[`setInterval`](https://developer.mozilla.org/en-US/docs/Web/API/setInterval), a
+browser function which repeatedly calls a function with a fixed delay between
+each call.
+
+Define a method `tickToNextWord` that will increment `this.idx`.
+
+{% switchable-sample %}
+
+```ts
+// word-viewer.ts
+
+  render() {
+    ...
+  }
+
+  tickToNextWord = () => { this.idx += 1; };
+
+```
+
+```js
+// word-viewer.js
+
+  render() {
+    ...
+  }
+
+  tickToNextWord = () => { this.idx += 1; };
+
+```
+
+{% endswitchable-sample %}
+
+Calling this method increments our reactive internal state `idx` which will
+schedule an update.
+
+To setup the interval and clean up the interval, use the [standard custom
+element lifecycle
+callbacks](/docs/components/lifecycle/#custom-element-lifecycle)
+`connectedCallback`, and `disconnectedCallback`.
+
+`connectedCallback` is invoked when the component is added to the DOM, and
+`disconnectedCallback` is invoked when the component is removed from the DOM.
+
+{% switchable-sample %}
+
+```ts
+// word-viewer.ts
+
+  private intervalTimer?: number;
+  
+  connectedCallback() {
+    super.connectedCallback();
+    this.intervalTimer = setInterval(this.tickToNextWord, 1000);
+  }
+  
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    clearInterval(this.intervalTimer);
+    this.intervalTimer = undefined;
+  }
+
+```
+
+```js
+// word-viewer.js
+
+  intervalTimer;
+  
+  connectedCallback() {
+    super.connectedCallback();
+    this.intervalTimer = setInterval(this.tickToNextWord, 1000);
+  }
+  
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    clearInterval(this.intervalTimer);
+    this.intervalTimer = undefined;
+  }
+
+
+```
+
+{% endswitchable-sample %}
+
+When the `word-viewer` is attached to the DOM, `connectedCallback` is invoked
+and `setInterval` repeatedly calls `this.tickToNextWord` every 1000ms, or every
+second.
+
+Store the interval ID on `this.intervalTimer` which does not need to be reactive.
+
+When `word-viewer` is removed from the DOM, `disconnectedCallback` is invoked
+cleaning up the interval, and `this.tickToNextWord` is no longer called.
```

Comment:
Perhaps an extra emphasis here to not forget this step to prevent memory leaks?

Reply:
Spent a bit thinking about this and not sure how else to phrase - since I elude to this with "cleaning up the interval".

Do you have a phrasing suggestion?

Reply:
You're right that "cleaning up" here already covers that. I can't think of anything more to say without it being redundant. Please disregard this comment!